set-static-ip

```diff
@@ -8,34 +8,57 @@ fi
 if [ -z "$PROVISIONING_INTERFACE" ]; then
```

Comment:
nit: move common code to a shared location

Reply:
Can potentially do a follow-up with this.